grc/submit_and_pay/__init__.py

```diff
@@ -271,3 +281,13 @@ def get_previous_page(application_data: ApplicationData, previous_page_in_journe
         section_check_your_answers_page=None,
         section_status=application_data.section_status_submit_and_pay_data,
         application_data=application_data)
+
+
+def anonymise_application(application_to_anonymise):
+    ApplicationFiles().delete_application_files(
+        application_to_anonymise.reference_number,
+        application_to_anonymise.application_data(),
+    )
+    application_to_anonymise.email = ''
+    application_to_anonymise.user_input = ''
+    application_to_anonymise.status = ApplicationStatus.ABANDONED
```

Comment:
Hi @alistairknight,
I reckon the bug is caused because you're not saving these changes to the database.
You'll need this line (either here or above after the for loop):
```python
db.session.commit()
```